sympy/polys/numberfields/tests/test_galoisgroups.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for computing Galois groups. """
+
+from sympy.combinatorics.named_groups import (
+    SymmetricGroup, AlternatingGroup, DihedralGroup, CyclicGroup,
+    AbelianGroup,
+)
+from sympy.combinatorics.permutations import Permutation
+from sympy.core.singleton import S
+from sympy.core.symbol import symbols
+from sympy.polys.domains.rationalfield import QQ
+from sympy.polys.numberfields.galoisgroups import (
+    Resolvent, tschirnhausen_transformation,
+    MaxTriesException, galois_group,
+    _galois_group_degree_4_simple,
+    M20, G36minus,
+)
+from sympy.polys.numberfields.subfield import field_isomorphism
+from sympy.polys.polytools import Poly
+from sympy.testing.pytest import raises, slow
+
+
+def test_Resolvent_roots():
+    X = symbols('X0 X1 X2 X3')
+    F = X[0]*X[2] + X[1]*X[3]
+    s = [
+        Permutation([0, 1, 2, 3]),
+        Permutation([1, 0, 2, 3]),
+        Permutation([3, 1, 2, 0])
+    ]
+    R = Resolvent(F, X, s)
+    roots = [r(*X) for r in R.root_lambdas]
+    assert roots == [
+        S('X0*X2 + X1*X3'),
+        S('X0*X3 + X1*X2'),
+        S('X0*X1 + X2*X3')
+    ]
+
+
+def test_tschirnhausen_transformation():
+    x = symbols('x')
+    for T in [
+        Poly(x**2 - 2),
+        Poly(x**2 + x + 1),
+        Poly(x**4 + 1),
+        Poly(x**4 - x**3 + x**2 - x + 1),
+    ]:
+        try:
+            _, U = tschirnhausen_transformation(T)
+            assert U.degree() == T.degree()
+            assert U.is_monic
+            assert U.is_irreducible
+            K = QQ.alg_field_from_poly(T)
+            L = QQ.alg_field_from_poly(U)
+            assert field_isomorphism(K.ext, L.ext) is not None
+        except MaxTriesException:
+            print('Max tries exceeded on Tschirnhausen transf.')
```

Comment:
If this is only catching an exception from the first line then only the first line should be in the try block.

Reply:
Also test code shouldn't print anything. It should either raise or pass silently.

Reply:
I eliminated the non-determinism, so the `try-except` is gone.